pkg/utils/k8sutil.go

```diff
@@ -387,12 +387,20 @@ func addInitContainerAnnotation(dpm *v1beta1.Deployment) error {
 
 // CreateIngress creates ingress rule for a specific function
 func CreateIngress(client kubernetes.Interface, funcObj *spec.Function, ingressName, hostname, ns string, enableTLSAcme bool) error {
-
 	or, err := GetOwnerReference(funcObj)
 	if err != nil {
 		return err
 	}
 
+	if len(funcObj.Spec.ServiceSpec.Ports) == 0 {
+		return fmt.Errorf("can't create route due to service port isn't defined")
+	}
+
+	port := funcObj.Spec.ServiceSpec.Ports[0].TargetPort
+	if port.IntVal <= 0 || port.IntVal > 65535 {
```

Comment:
you don't need to do this check, you can delegate the validation to the kubernetes library